gateway/1.4/modules/ROOT/pages/flex-gateway-k8-high-availability.adoc

```diff
@@ -0,0 +1,286 @@
+TEST SUITE: None
+----
+
+. Verify your autoscaling configuration by running the following command:
++
+.Syntax:
+[source, kubectl]
+----
+kubectl get hpa -n <namespace>
+----
++
+.Example:
+[source, kubectl]
+----
+kubectl get hpa -n gateway
+----
++
+The command prints output similar to this example: 
++
+----
+NAME     REFERENCE           TARGETS          MINPODS  MAXPODS  REPLICAS  AGE
+ingress  Deployment/ingress  66%/50%, 5%/50%  2        11       2         59s
+----
+//TODO: TIME PERMITTING, IT WOULD BE GOOD TO ADD SOME TROUBLESHOOTING INFO
+
+[[autoscaling-parameters]]
+=== Autoscaling Parameters
+
+To use autoscaling, enable Horizontal Pod Autoscaler (HPA), and configure other Helm chart settings to define autoscaling behavior.
+
+[%header,cols="1a,2a"]
+|===
+| Parameter | Description
+| `autoscaling.enabled` | Boolean that indicates whether the Horizontal Pod Autoscaler (HPA) is enabled. Defaults to `false`.
+| `autoscaling.minReplicas` | The minimum number of replicas that the HPA scaler is allowed to create. Defaults to `2`.
+| `autoscaling.maxReplicas` | The maximum number of replicas that the HPA scaler is allowed to create. Defaults to `11`.
+| `autoscaling.targetCPUUtilizationPercentage` | A resource metric that sets the average CPU usage percentage of all deployed Pods. Defaults to `50`.
+| `autoscaling.targetMemoryUtilizationPercentage` | A string representing the average memory usage percentage of all deployed Pods. Defaults to `nil`.
+| `autoscaling.extraMetrics` | A list of custom Pod configuration metrics. When a Pod runs with these settings, you can collect and monitor application-specific metric data. Defaults to `[]`. Define the metrics with nested parameters, for example:
+
+----
+extraMetrics:
+- type: Resource
+  resource:
+    name: cpu
+    target:
+      type: Utilization
+      averageUtilization: 50
+----
+
+You can retrieve metrics data with the following commands:
+
+----
+kubectl get --raw /apis/metrics.k8s.io/v1beta1/nodes/<node-name>
+
+kubectl get --raw /apis/metrics.k8s.io/v1beta1/namespaces/<namespace>/pods/<pod-name>
+----
+| `autoscaling.behavior` | A setting that supports the `HorizontalPodAutoscaler` (HPA) object in Kubernetes. HPA settings control autoscaling behavior, such as the number of application resources based on the rate of change to the workload. 
+| `autoscaling.behavior.scaleUp`, `autoscaling.behavior.scaleDown` | Settings for autoscaling behavior when the HPA scaler increases (`scaleUp`) or decreases (`scaleDown`) the number of replicas. Define scaling behavior with nested parameters: 
+
+* `selectPolicy`: If a metric of configured resources indicates that scaling is require, this setting determines when to scale Pods. This setting is necessary because Pods can contain different numbers of replicas. Valid values are `Disabled`, `Min`, and `Max`. Defaults to `Max`. The `Min` and `Max` values indicate whether to scale Pods with the lowest (`Min`) or highest (`Max`) number of replicas.
```

Comment:
```suggestion
* `selectPolicy`: If a metric of configured resources indicates that scaling is required, this setting determines when to scale Pods. This setting is necessary because Pods can contain different numbers of replicas. Valid values are `Disabled`, `Min`, and `Max`. Defaults to `Max`. The `Min` and `Max` values indicate whether to scale Pods with the lowest (`Min`) or highest (`Max`) number of replicas.
```